lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestContextStart.cs

```diff
@@ -0,0 +1,238 @@
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<ContextStartDriver>.Class)
+                .Set(DriverConfiguration.OnContextClosed, GenericType<ContextStartDriver>.Class)
+                .Build();
+        }
+
+        private sealed class ContextStartDriver :
+             IObserver<IDriverStarted>,
+             IObserver<IAllocatedEvaluator>,
+             IObserver<IActiveContext>,
+             IObserver<ICompletedTask>,
+             IObserver<IClosedContext>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+            private const string ContextId1 = "ContextID1";
+            private const string ContextId2 = "ContextID2";
+            private const string TaskId = "TaskID";
+            private bool _first = true;
+
+            [Inject]
+            private ContextStartDriver(IEvaluatorRequestor evaluatorRequestor)
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {               
+                Logger.Log(Level.Info, "IActiveContext: " + value.Id);
+
+                if (_first)
+                {
+                    Assert.Equal(value.Id, ContextId1);
+                    _first = false;
+                    value.SubmitContext(
+                        ContextConfiguration.ConfigurationModule
+                            .Set(ContextConfiguration.Identifier, ContextId2)
+                            .Set(ContextConfiguration.OnContextStart, GenericType<ContextStartHandler>.Class)
+                            .Build());
+                }
+                else
+                {
+                    Assert.Equal(value.Id, ContextId2);
+                    var c = TaskConfiguration.ConfigurationModule
+                        .Set(TaskConfiguration.Identifier, TaskId)
+                        .Set(TaskConfiguration.Task, GenericType<TestTask>.Class)
+                        .Build();
+                    value.SubmitTask(c);
+                }
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, ContextId1)
+                        .Set(ContextConfiguration.OnContextStart, GenericType<ContextStartHandler>.Class)
+                        .Build());
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                Logger.Log(Level.Info, "Task is completed:" + value.Id);
+                Assert.Equal(value.Id, TaskId);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IClosedContext value)
```

Comment:
I tested this and the test passes. This is bad news, as it shouldn't pass. Or am I missing something?